src/loaders/compute_tools/tool_result_parser.py

```diff
@@ -0,0 +1,105 @@
+import copy
+import os
+from pathlib import Path
+from typing import Dict, List, Set
+
+import jsonlines
+import pandas as pd
+
+from src.loaders.compute_tools.tool_common import GenomeTuple
+
+# used as collection_and_field_names.FLD_KBASE_ID
+TOOL_KBASE_ID = "kbase_id"
+TOOL_GENOME_ATTRI_FILE = "genome_attribs.jsonl"
+
+
+def create_jsonl_files(
+        file_path: Path,
+        docs: list) -> None:
+    """
+    create jsonl file from the list of documents
+
+    :param file_path: the path to the jsonl file
+    :param docs: the list of documents
+    """
+    print(f'Creating JSONLines import file: {file_path}')
+    with jsonlines.open(file_path, mode='w') as writer:
+        writer.write_all(docs)
+
+
+def read_genome_attri_result(
+        batch_result_dir: Path,
+        tool_file_name: str,
+        features: Set[str],
+        genome_id_col: str,
+        ids_to_files: Dict[str, GenomeTuple],
+        prefix: str = '') -> List[Dict[str, str]]:
+    """
+    process the output file generated by the tool (checkm2, gtdb-tk, etc) to create a format suitable for
+    importing into ArangoDB
+
+    :param batch_result_dir: the directory where the tool result files are stored
+    :param tool_file_name: the name of the tool result file
+    :param features: a list of features to be retrieved from the tool result file
+    :param genome_id_col: the column name of the genome id in the tool result file
+    :param ids_to_files: a dictionary of tool genome ids to the corresponding GenomeTuple
+    :param prefix: the prefix to append to the genome id from the tool result file
+
+    :return: a dictionary of genome attributes
+    """
+    tool_genome_map = {tool_id: genome_tuple.data_id for tool_id, genome_tuple in ids_to_files.items()}
+
+    tool_file = os.path.join(batch_result_dir, tool_file_name)
+    docs = list()
+    if os.path.exists(tool_file):
+        df = _read_tsv_as_df(tool_file, features, genome_id_col=genome_id_col)
+        docs = df.apply(_row_to_doc, args=(features, tool_genome_map,
+                                           genome_id_col, prefix), axis=1).to_list()
+        docs = [doc for doc in docs if doc]
+
+    return docs
+
+
+def _row_to_doc(row, features, tool_genome_map, genome_id_col, prefix):
+    # Transforms a row from tool result file into ArangoDB collection document
+
+    try:
+        genome_id = tool_genome_map[row[genome_id_col]]
+    except KeyError as e:
+        raise ValueError('Unable to find genome ID') from e
```

Comment:
When can this happen?

Reply:
hmmm maybe it's just a safe check. would you rather not having this? It was inherited from the parse_tool_result.py too.

Reply:
If we're sure it can't happen we should remove it, but if we're not sure it seems helpful to keep it

Reply:
it doesn't hurt. I think I will just keep it.

Reply:
:+1: